src/services/ConfigApi.js

```diff
@@ -74,6 +74,34 @@ export class ConfigApi {
 		}
 	}
 
+	async getBadgeTemplates() {
+		const apiPath = `${this.host}/api/badgeTemplates`;
+
+		const response = await fetch(apiPath, {
+			method: 'GET',
+			headers: {
+				Accept: 'application/json',
+				Authorization: this.secretKey,
+				'User-Agent': this.userAgent,
+			},
+		});
+
+		if (response.status == 200) {
+			return await response.json();
+		} else if (response.status == 401) {
+			throw new Error(`Invalid secretKey.`);
+		} else if (response.status == 405) {
+			throw new Error(`Server method not allowed.`);
+		} else if (response.status == 429) {
+			const text = (await response.text()).trim();
+			throw new Error(`Try again: ${text}`);
+		} else if (response.status == 500) {
+			throw new Error(`Server encounterd a problem.`);
+		} else {
+			throw new Error(`Unknown error has occured.`);
+		}
```

Comment:
Not super important, but any way we could DRY this up a bit?

Reply:
I see that this might not be possible due to wanting to respond differently for each endpoint, but maybe there is some clever way?